arch/risc-v/src/common/riscv_internal.h

```diff
@@ -335,6 +351,7 @@ void riscv_sbi_set_timer(uint64_t stime_value);
 uint64_t riscv_sbi_get_time(void);
 uintptr_t riscv_sbi_boot_secondary(uint32_t hartid, uintptr_t addr,
                                    uintptr_t a1);
+uintptr_t riscv_sbi_system_reset(uint32_t type, uint32_t reason);
```

Comment:
what about non-S mode?

Reply:
There seems no standard way to do that in non-S mode. For `rv-virt` target, more investigation is needed to see how to do it in M-mode later.

Reply:
OK